File: backend/app/schemas/excel_io.py
from pydantic import BaseModel, Field


class ExcelImportErrorRead(BaseModel):
    row: int
    field: str
    message: str


class ExcelImportResultRead(BaseModel):
    total_rows: int = 0
    created_count: int = 0
    updated_count: int = 0
    skipped_count: int = 0
    errors: list[ExcelImportErrorRead] = Field(default_factory=list)

File: frontend/src/types/excel-io.ts
export type ImportKind = "projects" | "centers" | "subjects" | "stage-templates";

export type ImportError = {
  row: number;
  field: string;
  message: string;
};

export type ImportResult = {
  total_rows: number;
  created_count: number;
  updated_count: number;
  skipped_count: number;
  errors: ImportError[];
};

export type ExportKind =
  | "project-progress"
  | "center-status"
  | "subject-completeness"
  | "missing-items";
